server/v2/cometbft/abci.go

```diff
@@ -104,6 +106,11 @@ func (c *Consensus[T]) SetStreamingManager(sm streaming.Manager) {
 	c.streaming = sm
 }
 
+// SetListener sets the listener for the consensus module.
```

Comment:
I don't like to add setters tbh, I feel like it pollutes the server instantiation in root.go
Additionally, Consensus is wired from the cometbft server and not manually.
Let's instead add it to the Consensus contructor if possible and have the CometBFT ServerOptions take a listener.

Reply:
That makes sense, the options can be nil by default then. But a user won't have to call NewConsensus manually, so putting it in a setter isn't useful

Reply:
works too, could you wire that in cometbft server.go?